src/rust/engine/ui/src/console_ui.rs

```diff
@@ -118,11 +120,17 @@ impl ConsoleUI {
     // render might barely miss a data refresh.
     let draw_target = ProgressDrawTarget::term(term, Self::render_rate_hz() * 2);
     let multi_progress = MultiProgress::with_draw_target(draw_target);
-
-    let bars = (0..self.local_parallelism)
+    let terminal_dimensions = terminal_size().unwrap_or((
+      Width(50),
+      Height(self.local_parallelism.try_into().unwrap()),
+    ));
+    let terminal_width = terminal_dimensions.0 .0;
+    let terminal_height = terminal_dimensions.1 .0 - 1;
```

Comment:
Why `-1` here? I think that this means that we'll render one fewer bar than there is concurrency.

Reply:
Maybe you should only subtract if we successfully detected the concurrency (i.e., put it in a `map` after the detection): see suggestion.

Reply:
Good point!

Reply:
Good point!